go/vt/vtgate/vindexes/cfc_test.go

```diff
@@ -30,94 +30,117 @@ import (
-		},
+func TestCFCCreateVindex(t *testing.T) {
+	cases := []createVindexTestCase{
+		cfcCreateVindexTestCase(
+			"no params",
+			nil,
+			nil,
+			nil,
+		),
+		cfcCreateVindexTestCase(
+			"no hash",
+			map[string]string{},
+			nil,
+			nil,
+		),
+		cfcCreateVindexTestCase(
+			"no hash with offsets",
+			map[string]string{"offsets": "[1,2]"},
+			nil,
+			nil,
+		),
+		cfcCreateVindexTestCase(
+			"hash with no offsets",
+			map[string]string{"hash": "md5"},
+			vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "CFC vindex requires offsets when hash is defined"),
+			nil,
+		),
+		cfcCreateVindexTestCase(
+			"invalid offsets 10,12",
+			map[string]string{"hash": "md5", "offsets": "10,12"},
+			vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "invalid offsets 10,12 to CFC vindex cfc. expected sorted positive ints in brackets"),
+			nil,
+		),
+		cfcCreateVindexTestCase(
+			"invalid offsets xxx",
+			map[string]string{"hash": "md5", "offsets": "xxx"},
+			vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "invalid offsets xxx to CFC vindex cfc. expected sorted positive ints in brackets"),
+			nil,
+		),
+		cfcCreateVindexTestCase(
+			"empty offsets",
+			map[string]string{"hash": "md5", "offsets": "[]"},
+			vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "invalid offsets [] to CFC vindex cfc. expected sorted positive ints in brackets"),
+			nil,
+		),
+		cfcCreateVindexTestCase(
+			"unsorted offsets",
+			map[string]string{"hash": "md5", "offsets": "[10,3]"},
+			vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "invalid offsets [10,3] to CFC vindex cfc. expected sorted positive ints in brackets"),
+			nil,
+		),
+		cfcCreateVindexTestCase(
+			"negative offsets",
+			map[string]string{"hash": "md5", "offsets": "[-1,3]"},
+			vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "invalid offsets [-1,3] to CFC vindex cfc. expected sorted positive ints in brackets"),
+			nil,
+		),
+		cfcCreateVindexTestCase(
+			"duplicated offsets",
+			map[string]string{"hash": "md5", "offsets": "[4,4,6]"},
+			vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "invalid offsets [4,4,6] to CFC vindex cfc. expected sorted positive ints in brackets"),
+			nil,
+		),
+		cfcCreateVindexTestCase(
+			"unknown params",
+			map[string]string{"hash": "md5", "offsets": "[3, 7]", "hello": "world"},
+			nil,
+			[]VindexWarning{vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "unknown param 'hello'")},
+		),
 	}
 
-	for _, tc := range cases {
-		t.Run(tc.testName, func(t *testing.T) {
-			cfc, err := NewCFC("cfc", tc.params)
-			assertEqualVtError(t, tc.err, err)
-			if cfc != nil {
-				assert.EqualValues(t, tc.offsets, cfc.(*CFC).offsets)
-				assert.Equal(t, "cfc", cfc.String())
-				assert.Equal(t, 1, cfc.Cost())
-				assert.Equal(t, true, cfc.IsUnique())
-				assert.Equal(t, false, cfc.NeedsVCursor())
-			}
-		})
-	}
+	testCreateVindexes(t, cases)
+}
+
+func TestCFCCreateVindexOptions(t *testing.T) {
```

Comment:
This test case was extracted from `TestCFCBuildCFC`. It was the only case that didn't inspected internal struct params, and didn't fit the generalized `testCreateVindexes` framework.